NavBar2 (front-end)

Khal-Shah opened the pull request:
![image](https://user-images.githubusercontent.com/35305423/36245129-a04fa9aa-11f7-11e8-8f28-6d97f1649fe0.png)

From what I gather, this needs further work involving user sessions/authentications for the texts and buttons to change when user is signed in.

When back end is fully implemented and there is a helper method that can be used to check if a user is signed in, the ask question button should turn green and redirect to new question page and the sign in text on NavBar should change to sign out.

I looked around but couldn't figure out how to get the back end done, but we need something like section 8.2.4 here:
http://rails-3-2.railstutorial.org/book/sign_in_out

Modified files: application.scss and application.html.erb (as navbar is supposed to be present on every page). Also put yield in container to center the content of page.

Khal-Shah committed:
NavBar Version 2

Khal-Shah committed:
NavBar front design

Khal-Shah committed:
NavBar front end, needs backend to check if user is online to be fully functional

Khal-Shah committed:
changed asked question button path to sign in page when user is not signed in

samgo1 commented:
@ThierrySt-Arnaud , I put you as reviewer because the navbar has merge conflicts with the user panel stuff, can you check that out and resolve conflicts? When everything looks good you can merge it to dev

ThierrySt-Arnaud commented on QA-webapp/app/assets/stylesheets/application.scss:
```diff
@@ -6,3 +6,43 @@
 @import "users";
 @import "welcome";
 @import "userpanel";
+
```
Everything after this should be put into another file (say navbar.scss) and then an @import statement should be added here (if the file is navbar.scss, then it should be `@import "navbar";`). This way you can modify all the css for the navbar in the one place without potentially breaking the rest and changing the order for the css file is much easier.

ThierrySt-Arnaud reviewed (changes requested):
I changed the order of a few lines, but otherwise everything should work as is. Just one thing I would change. Its nitpicking but I think it will save us a lot of trouble down the line; we can easily separate the css files so that we don't have all the css in one file that will constantly be conflicting as we modify it. I put instructions in a comment.

Khal-Shah committed:
changes accompanying the merge

Khal-Shah committed:
merge changes

Khal-Shah committed:
Updated Navbar, currently semi-functional

Khal-Shah commented:
I put the styling in a different file as requested. 

Fixed the padding issue, changed background color. 

The Sign In button does change to Sign Out once an user is created, but I still couldn't find the delete session path so the sign out button currently gets routed to home page when user signs out.

The question button changes to green from red upon user creation and routes to new question page when user clicks on it.
![image](https://user-images.githubusercontent.com/35305423/36301533-3948cc5a-12d3-11e8-88be-0564c7f315cb.png)

What still needs to be fixed: once user clicks on the green question button, it changes back to red in the new question page. Same for the Sign Out button (changes back to Sign In).

Also I still get an InvalidAuthenticityToken error when trying to post a question.

I haven't added a "List Questions" item on the navbar, I can try to do that tomorrow given we can make what we currently have functional first. Otherwise, that seems more like a Sprint 3 thing.

samgo1 reviewed (changes requested):
We don't need the additional database entity attributes for the question and we don't need a 'views' table in the db. Can you remove the addition from QA-webapp/db/schema.rb?

ThierrySt-Arnaud reviewed (approved):
Everything looks good for me.

Khal-Shah committed:
without schema.rb changes

Khal-Shah commented:
I'm not sure how those got into there...But I removed everything new that were added to schema.rb

Khal-Shah committed:
added show questions button

Khal-Shah commented:
Added a static show question button that shows the listing of questions at /questions:
![image](https://user-images.githubusercontent.com/35305423/36317229-8debb470-130a-11e8-9daf-600fa0b58d07.png)

samgo1 committed:
Merge branch 'dev' into NavBar2
